src/features/situation-builder/components/CustomButton.tsx

```
@@ -0,0 +1,97 @@
+import { Button } from "@/components/ui/button";
+import clsx from "clsx";
+import { Loader2, WandSparkles } from "lucide-react";
+import React from "react";
+
+interface SubmitButtonProps {
+  isLoading?: boolean;
+  children: React.ReactNode;
+  className?: string;
+  [key: string]: any;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)